tests/integration/test_local_variables.py

```diff
@@ -558,3 +558,71 @@ def second_func(the_argument):
 
     second_func_frame = find_frame(frames, "second_func")
     assert "the_argument" not in second_func_frame.arguments
+
+
+@ALL_PYTHONS
+@ALL_SOURCES
+def test_trashed_locals(generate_threads, python, tmpdir):
+    # GIVEN
+    _, python_executable = python
+
+    program_template = """
+import ctypes
+import sys
+import time
+
+class ListObject(ctypes.Structure):
```

Comment:
I am a bit hesitant because this test may change if we change/optimise the internal representation of these objects, which is not likely but also it's not extremely unlikely.

Reply:
I considered that, but we already have [src/pystack/_pystack/cpython/list.h](https://github.com/bloomberg/pystack/blob/456f12bb9464e5dc38026fda0d9ed4a0271de019/src/pystack/_pystack/cpython/list.h) and [src/pystack/_pystack/cpython/tuple.h](https://github.com/bloomberg/pystack/blob/456f12bb9464e5dc38026fda0d9ed4a0271de019/src/pystack/_pystack/cpython/tuple.h) which will already need to be updated if CPython ever changes or if we ever start to support builds with `-DPy_TRACE_REFS`. This just adds one more place we'll need to update, but it's no more fragile than what the tool itself already does.

Reply:
In any event, if we ever need to, we can test this with a custom C extension that trashes types that we own and can control the layout of. ctypes is a quick and dirty way to do it from pure Python, but these changes aren't specific to `list` and `tuple`, and we could instead test them with our own extension types with a bit more work if we ever have a reason to.